### PR TITLE
Bug fixing - First Vehicle Spawned by player

### PR DESCRIPTION
**Pull request by CaiusAjiz**
Fixes an issue where the first vehicle or a type (MATV, Truck etc...) requested by a player is undrivable for them. Appears to come from where the createVehicle used to spawn on 0,0,0 then moved to the marked location. 

This has been removed and the vehicle now spawns directly on the requested location. No noticable performace hit.

Tested on SP, MP and Dedi.

### CODE REVIEW

**Comment by CaiusAjiz** [comment]
Pull #1 Will need to go in before this. I'll then re-test the above as #1 changes some of the files that this works on.

**Comment by amanuense** [comment]
the reason to spawn vehicles in 0,0,0 is a recommendation I found here
https://community.bistudio.com/wiki/Code_Optimisation
if that fixes the issue I don't mind having that change

File: source/dialog/request_vehicle.sqf [hunk 1]
```diff
@@ -39,27 +39,27 @@ _spawnPos = [_spawnPos, 20, 50, 3, 0, 1, 0] call BIS_fnc_findSafePos;
 
 
 switch (_index) do {
-	case 0: { _vehic = [_spawnPos, "I_G_Offroad_01_F", offroad_cost] call spawn_vehicle_fn; };
```
Comment (amanuense) [inline source/dialog/request_vehicle.sqf:42]: i find the change of variable name unnecessary. it's ok to make the change but remember that when integrating the change that can add to the possibility of having merge issues

File: source/support/spawn_vehicle.sqf [hunk 2]
```diff
@@ -21,11 +21,9 @@ if (commandpointsblu1 >= _cost) then {
 	hint "Vehicle ready! check map for location";
 	commandpointsblu1 = commandpointsblu1 - _cost;
 	ctrlSetText [1000, format["%1",commandpointsblu1]];
-	_vehic = _name createVehicle [0,0,0];
```
Comment (amanuense) [inline source/support/spawn_vehicle.sqf:24]: according to https://community.bistudio.com/wiki/Code_Optimisation
createVehicle(Local)
createVehicle(Local) position is not exact so you must use setPos but this is very slow, to create the object on [0,0,0] and then set the position is faster.

_obj = 'Land_Stone_4m_F' createVehicle [0,0,0]; //also createVehicleLocal
_obj setPos (getPos player); //0,03ms (100 testcycles)
is 200 times faster than...

_obj = 'Land_Stone_4m_F' createVehicle (getPos player); //also createVehicleLocal
_obj setPos (getPos player); //5,9ms (100 testcycles)

Comment (amanuense) [inline source/support/spawn_vehicle.sqf:24]: if the change really fixes the issue, it'll be ok to have it, we don't spawn dozens of vehicles anyways

**Review by amanuense — COMMENTED** [review]
left some feedback

**Comment by CaiusAjiz** [comment]
I'm going to close this request off, I'll take the suggestions mentioned and continue with some more testing, as ideally if we can keep the createVehicle [0,0,0] it help with speed. The variable rename was something i missed too.

I'm not satisfied with the amount of testing i did for this so i'll revisit this after spending more time on it. Thanks for the suggetions amanuense!

**Pull request closed without merging**